src/pages/ValidationPage.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable react/jsx-indent */
+/* eslint-disable react/jsx-closing-tag-location */
+/* eslint-disable indent */
+import React, { useEffect, useState } from "react";
+import { CardContent, CircularProgress, makeStyles, Button } from "@material-ui/core";
+import { useMobile } from "../utils/media-queries";
+import { Redirect, useParams } from "react-router-dom";
+import { validateApplication } from "../services/companyApplicationService";
+import { getValidationError } from "../components/Apply/Company/CompanyApplicationUtils.js";
+
+const useStyles = (isMobile) => makeStyles((theme) => ({
+    content: {
+        padding: isMobile ? theme.spacing(2, 2) : theme.spacing(3, 3),
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "center",
+    },
+    title: {
+        color: "rgb(80,80,80)",
+        fontSize: "23px",
+    },
+    text: {
+        color: "rgb(100,100,100)",
+        fontSize: "15px",
+        maxWidth: isMobile ? "100%" : "80%",
+        textAlign: "center",
+
+    },
+    button: {
+        background: "rgb(250,70,70)",
+        color: "white",
+        marginTop: "3vh",
+
+    },
+}));
+
+const ValidationPage = () => {
+    const isMobile = useMobile();
+    const classes = useStyles(isMobile)();
+    const { token } = useParams();
+    const [loading, setLoading] = useState(true);
+    const [success, setSuccess] = useState(false);
+    const [buttonPressed, setButtonPressed] = useState(false);
+    const [error, setError] = useState("");
+
+
+    useEffect(() => {
+        async function validate() {
+            try {
+                setLoading(false);
+                setSuccess(true);
+                await validateApplication(token);
+            } catch (error_) {
+                setError(error_[0].msg);
+                setLoading(false);
+                setSuccess(false);
+            }
+
+        }
```

Comment:
Why are you creating a method and calling it immediately after?

Reply:
UseEffect can't be async, because of that if you want to have an await inside of an useEffect you have to use an auxiliar async function , .then() would also work as an alternative